cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -467,13 +651,13 @@ public ProgramRunStatus recordProgramSuspend(ProgramRunId programRunId, byte[] s
    */
   @Nullable
   public ProgramRunStatus recordProgramResumed(ProgramRunId programRunId, byte[] sourceId) {
-    RunRecordMeta existing = getRunRecord(programRunId);
-    // ProgramRunStatus.RUNNING will actually be persisted but ProgramRunStatus.RESUMING is used here to distinguish
-    // recordProgramResumed from recordProgramRunning, since the two methods have different sets of allowed statuses
-    // of the existing records
-    boolean isValid = validateExistingRecords(existing, programRunId, sourceId,
-                                              "resume", ProgramRunStatus.RESUMING);
-    if (!isValid && existing != null) {
+    RunRecordMeta existing = getRun(programRunId);
+    if (existing == null) {
+      LOG.warn("Ignoring unexpected transition of program run {} to program state {} with no existing run record.",
+               programRunId, ProgramRunStatus.RUNNING);
```

Comment:
Do you mean: `ProgramRunStatus.RESUMING`?

Reply:
oops, right

Reply:
this is actually correct. The current code has a confusing implementation where there is a 'resuming' status that is not actually a program state, but is used as an action to transition between states.

Reply:
I see.